aptos-move/block-executor/src/view.rs

```diff
@@ -107,13 +107,38 @@ impl<'a, T: Transaction, X: Executable> ParallelState<'a, T, X> {
             .set_base_value(id, base_value)
     }
 
-    fn read_aggregator_v2_committed_value(
+    fn read_aggregator_v2_last_committed_value(
         &self,
         id: T::Identifier,
+        txn_idx: TxnIndex,
+        read_position: ReadPosition,
     ) -> Result<AggregatorValue, MVAggregatorsError> {
         self.versioned_map
             .aggregators()
-            .read_latest_committed_value(id)
+            .read_latest_committed_value(id, txn_idx, read_position)
+    }
+
+    fn read_aggregator_v2_aggregated_value(
+        &self,
+        id: T::Identifier,
+        txn_idx: TxnIndex,
+    ) -> Result<AggregatorValue, MVAggregatorsError> {
+        match self.versioned_map.aggregators().read(id, txn_idx) {
+            Ok(value) => {
+                if self
+                    .captured_reads
+                    .borrow_mut()
+                    .capture_delayed_field_read(id, value.clone())
+                    .is_err()
+                {
+                    // Inconsistency in recorded reads.
+                    Err(MVAggregatorsError::DeltaApplicationFailure)
```

Comment:
We can err here in both speculative and wrong use scenarios (continuing from captured_reads.rs)

Reply:
follow-up PR distinguishes that